### PR TITLE
Replace the internal custom logger with Log4j standard loggers

### PR DESCRIPTION
**Pull request by diranged**
The primary work here is to remove the internal custom logging library
and replace it with a standard Log4j setup. This lets the user define
their own logging settings in a standard way that doesn't require a
bunch of custom settings in the AgentConfiguration class.

Unfortunately there are a few other fixes in here that we've squashed
down - I've tried to list them here:

* Fix a few bugs in the Ubuntu init script
* Dynamically find the 'service' command path in the babysit script
* Leverate the actual init scripts for auto-restarting the agent if necessary.
* A few more Ubuntu init script bug fixes.
* Package the app up into amazon-kinesis-agent-<tag>.tar.gz files.
* Bugfix: /etc/default, not /etc/defaults on Ubuntu
* Bugfix on the babysit script: --ppid doesn't work everywhere.
* Remove custom logger

### CODE REVIEW

**Comment by diranged** [comment]
@chaochenq,
  This is a cleanup that replaces #83 - we've gotten rid of our broken `master` branch fork and cleaned things up. This PR is based on the latest `awslabs:master`, and includes all of the necessary changes to get the agent up to where we've been running it for the last few months. This codebase is simpler because it removes any of the custom log4j settings and lets users just do what they need to with their own log4j configs. This allows for far more granular control of the logging, as well as integration with tools like our https://github.com/Nextdoor/rollbar-log4j module.

Could you review this and let us know if its OK? I really want to get this merged and then do a fast-followup with our fat-jar PR.

**Comment by chaochenq** [comment]
Noted, I'll make sure to look at this by the end of this week. Thanks!

**Comment by chaochenq** [comment]
@diranged sorry for the delay... there was some task churn on our team. I'm going to look at this today

File: src/com/amazon/kinesis/streaming/agent/config/AgentConfiguration.java [hunk 1]
```diff
@@ -49,10 +48,6 @@
     public static final String ASSUME_ROLE_EXTERNAL_ID = "assumeRoleExternalId";
     public static final String SHUTDOWN_TIMEOUT_MILLIS_KEY = "shutdownTimeoutMillis";
     public static final String ENDPOINT_KEY = "endpoint";
-    static final String LOG_FILE_KEY = "log.file";
```
[the remark this answers is not in the document]
Comment (diranged) [reply src/com/amazon/kinesis/streaming/agent/config/AgentConfiguration.java]: @chaochenq, 
 Sorry I missed this response.. your flurry of other PR merges got my attention! On this, I really think the "right" thing to do here is to intentionally release a version bump that with a big notification around the logging change. Implementing _both a config.json based logging config_ AND a _standard log4j config_ is really confusing and going to introduce a ton of if/else logic that doesn't need to be there. I feel pretty strongly that moving to a standard log4j configuration is the right thing to do here. It's a well known standard in the Java space, and it provides the most flexibility for your users to define their own logging parameters.

  Do you really think that you have any users of this tool who wouldn't be just as easily able to push a custom log4j file? I'd be surprised if that was the case.

File: bin/aws-kinesis-agent-babysit [hunk 2]
```diff
@@ -4,19 +4,30 @@
...
+fi
+
 function start_agent() {
   $SERVICE restart || exit 1
   sleep 3
   $SERVICE status >/dev/null 2>&1 || exit 1
 }
 
-$SERVICE status >/dev/null 2>&1
-status=$?
```
[the remark this answers is not in the document]
Comment (diranged) [reply bin/aws-kinesis-agent-babysit]: Indeed - an accident. Put them back.

File: src/com/amazon/kinesis/streaming/agent/config/AgentConfiguration.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (chaochenq) [reply src/com/amazon/kinesis/streaming/agent/config/AgentConfiguration.java]: Yea I agree. These changes look good to me and I have been wanting to do such a switch for a while too. Thanks for the changes. I can merge this soon and call it out in the release note.

[the remark this answers is not in the document]
Comment (diranged) [reply src/com/amazon/kinesis/streaming/agent/config/AgentConfiguration.java]: :+1: excellent, thanks! The next change (the big fat jar one) is an even bigger improvement in the build-ability and ship-ability of this app. we've been running it for weeks now and it has vastly simplified the deployment process..

File: setup [hunk 3]
```diff
@@ -196,7 +196,8 @@ do_install () {
   install -m755 ./bin/${daemon_name}.${dist} ${init_dir}/${daemon_name}
   install -m644 ./support/${daemon_name}.cron ${cron_dir}/${daemon_name}
   install -m644 ./support/${daemon_name}.sysconfig ${sysconfig_dir}/${daemon_name}
-  
+  install -m644 ./support/log4j.xml ${config_dir}
```
Comment (chaochenq) [inline setup]: where is this new ./support/log4j.xml file? I don't see it in this diff

File: src/com/amazon/kinesis/streaming/agent/config/AgentConfiguration.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (chaochenq) [reply src/com/amazon/kinesis/streaming/agent/config/AgentConfiguration.java]: I just added another comment. Where is the new log4j.xml? it's supposed to exist in support/ but I don't find it in this diff

[the remark this answers is not in the document]
Comment (chaochenq) [reply src/com/amazon/kinesis/streaming/agent/config/AgentConfiguration.java]: I'm sorry, my bad. I see it now.

File: setup (hunk 3 above)
Comment (chaochenq) [reply setup]: sorry my bad. I see it now.

**Comment by chaochenq** [comment]
I have merged the changes. But I'm holding off the release for now because I want to check some other features in and make the version bump to 1.2 at the same time. The estimated release date is end of Oct.

@diranged does that work for you?

**Comment by diranged** [comment]
Excellent - we're fine because we are running on our own fork anyways. However, we want to keep our fork as closely aligned with your codebase as possible.

**Pull request merged**